lib/Util.ts

```diff
@@ -49,4 +58,117 @@ export class Util {
+          `wif private key must be included in Compressed WIF format.`
+        )
+      }
+
+      // Input validation
+      if (!balanceOnly) {
+        if (!toAddr || toAddr === "") {
+          throw new Error(
+            `Address to receive swept funds must be included unless balanceOnly flag is true.`
+          )
+        }
+      }
+
+      // Generate a keypair from the WIF.
+      const keyPair = ecPair.fromWIF(wif)
+
+      // Generate the public address associated with the private key.
+      const fromAddr = ecPair.toCashAddress(keyPair)
+
+      // Check the BCH balance of that public address.
+      const details: any = await axios.get(
+        `${this.restURL}address/details/${fromAddr}`
+      )
+      const balance = details.data.balance
+
+      // If balance is zero, exit.
+      if(balance === 0) return balance
+
+      // If balanceOnly flag is passed in, exit.
+      if(balanceOnly) return balance
+
+      // Get UTXOs associated with public address.
+      const u: any = await axios.get(
+        `${this.restURL}address/utxo/${fromAddr}`
+      )
+      const utxos = u.data.utxos
+
+      // Prepare to generate a transaction to sweep funds.
+      const transactionBuilder = new TransactionBuilder()
+      let originalAmount = 0
+
+      // Add all UTXOs to the transaction inputs.
+      for (let i = 0; i < utxos.length; i++) {
+        const utxo = utxos[i]
+
+        originalAmount = originalAmount + utxo.satoshis
+
+        transactionBuilder.addInput(utxo.txid, utxo.vout)
+      }
+
+      if (originalAmount < 1)
+        throw new Error(`Original amount is zero. No BCH to send.`)
+
+      // get byte count to calculate fee. paying 1.1 sat/byte
+      const byteCount = bitcoinCash.getByteCount(
+        { P2PKH: utxos.length },
+        { P2PKH: 1 }
+      )
+      const fee = Math.ceil(1.1 * byteCount)
+
+      // amount to send to receiver. It's the original amount - 1 sat/byte for tx size
+      const sendAmount = originalAmount - fee
+
+      // add output w/ address and amount to send
+      transactionBuilder.addOutput(
+        addressLib.toLegacyAddress(toAddr),
+        sendAmount
+      )
+
+      // Loop through each input and sign it with the private key.
+      let redeemScript
+      for (var i = 0; i < utxos.length; i++) {
+        const utxo = utxos[i]
+
+        transactionBuilder.sign(
+          i,
+          keyPair,
+          redeemScript,
+          transactionBuilder.hashTypes.SIGHASH_ALL,
+          utxo.satoshis
+        )
+      }
+
+      // build tx
+      const tx = transactionBuilder.build()
+
+      // output rawhex
+      const hex = tx.toHex()
+
+      // Broadcast the transaction to the BCH network.
+      const response: any = await axios.get(
+        `${REST_URL}rawtransactions/sendRawTransaction/${hex}`
+      )
+
+      return response.data
+    } catch (error) {
+      if (error.response && error.response.data) throw error.response.data
+      else throw error
+    }
+  }
```

Comment:
The sweep function makes three calls to rest.bitcoin.com, which will make the unit tests clunky to mock.